src/note.ts

```diff
@@ -218,30 +464,30 @@ export class Note extends Tickable {
     }
 
     // The render surface.
-    this.stave = null;
     this.render_options = {
       annotation_spacing: 5,
     };
   }
 
   // Get and set the play note, which is arbitrary data that can be used by an
   // audio player.
-  getPlayNote() {
+  getPlayNote(): Note | undefined {
     return this.playNote;
   }
-  setPlayNote(note) {
+
+  setPlayNote(note: Note): this {
     this.playNote = note;
     return this;
   }
 
   // Don't play notes by default, call them rests. This is also used by things like
   // beams and dots for positioning.
-  isRest() {
+  isRest(): boolean {
     return false;
   }
 
   // TODO(0xfe): Why is this method here?
-  addStroke(index, stroke) {
+  addStroke(index: number, stroke: Stroke): this {
```

Comment:
@0xfe what is the reason for the TODO?

Reply:
Leave it for now. It was around whether it belongs in `Note` vs `StemmableNote`, because it's used just by `StaveNote` and `TabNote`.